modules/c++/numpyutils/source/numpyutils.cpp

```diff
@@ -159,10 +159,44 @@ void createOrVerify(PyObject*& pyObject,
     }
 }
 
-void prepareInputAndOutputArray(PyObject* pyInObject, 
-                                            PyObject*& pyOutObject, 
-                                            int inputTypeNum, 
-                                            int outputTypeNum, 
+PyObject* toNumpyArray(size_t numRows, size_t numColumns,
+        int typenum, void* data)
+{
+    int nDims = (numRows == 1 ? 1 : 2);
+    npy_intp dimensions[2];
+    if (nDims == 1)
+    {
+        dimensions[0] = numColumns;
+    }
+    else
+    {
+        dimensions[0] = numRows;
+        dimensions[1] = numColumns;
+    }
+    return PyArray_NewCopy((PyArrayObject*)
```

Comment:
Use C++-style `reinterpret_cast` rather than C-style cast here for `PyArrayObject`